File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        s = nums[0]
        if(s==0 and n>1):
            return False
        for i in range(1,n-1):
            s-=1
            if(nums[i]>s):
                s = nums[i]
            if(s<=0):
                return False
        return True

            


        

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int x = nums[0];
        if(nums.length == 1)
        {
            return true;
        }
        else if(x<=0)
        {
            return false;
        }

        for(int i=1;i<nums.length-1;i++)
        {
            x--;
            if(x<nums[i])
            {
                x = nums[i];
            }
            if(x<=0)
            {
                return false;
            }
        }
        return true;
        
    }
}